streamlit_drawable_canvas/frontend/src/DrawableCanvas.tsx

```diff
@@ -142,8 +145,11 @@ const DrawableCanvas = ({ args }: ComponentProps) => {
       strokeColor: strokeColor,
     })
 
-    canvas.on("mouse:up", () => {
+    canvas.on("mouse:up", (e: any) => {
       saveState(canvas.toJSON())
+      if (e["button"] === 3) {
+        forceStreamlitUpdate()
```

Comment:
I get a `Line 165:6:  React Hook useEffect has a missing dependency: 'forceStreamlitUpdate'. Either include it or remove the dependency array  react-hooks/exhaustive-deps` 

Can you add `forceStreamlitUpdate` to the array line 165? This is only to appease React, it should have no impact on anything XD

Reply:
(Just curious... Why I don't get the error at my side?)

I have fixed the typo in and add the line you mentioned and tested on my machine.

I forgot to mention one thing which has worried me a bit. In previous submission, I encountered a bug several times but I just could not repeat it. When I was trying to toggle "Update in realtime" on and off with all the drawing tools. Sometimes the canvas began to "vibrate" until I reload the page. I wondered that whether it related to any "infinite loop" or not (I remember you once reminded me about that but I don't understand it actually...). Later I tried to figure out what kind of operation may cause that bug, but I never be able to reproduce it again.

Reply:
> (Just curious... Why I don't get the error at my side?) 

It's more of a warning than an error that appears in the console when I run `npm run start` (and `VSCode` on my side is configured to display eslint problems too), it comes from an implicit eslint linting rule from the set of rules in `react-hooks/exhaustive-deps`. If you watch other Frontend projects, those rules are defined in a `.eslintrc`  file usually, and maybe I should have done it explicitly to ensure we have the same eslint rules applying.

> When I was trying to toggle "Update in realtime" on and off with all the drawing tools. Sometimes the canvas began to "vibrate" until I reload the page. I wondered that whether it related to any "infinite loop" or not (I remember you once reminded me about that but I don't understand it actually...). Later I tried to figure out what kind of operation may cause that bug, but I never be able to reproduce it again.

Yeah we have an issue for this on the Streamlit side, but because it's not reproducible it's just hard to debug...I say it's ok for now and not a lot of people are going to toggle this furiously